src/type-factor.c

```diff
@@ -1,6 +1,88 @@
 #include "vctrs.h"
 #include "utils.h"
 
+static SEXP vec_restore_bare_factor_impl(SEXP x, SEXP to, SEXP to_levels) {
+  const int to_max = vec_size(to_levels);
```

Comment:
```suggestion
  const R_len_t to_max = vec_size(to_levels);
```

Just for conformity of types if we swich vec-size to long sizes. This way we'll support the `factor(1:long_size)` case.